tests/conftest.py

```diff
@@ -69,18 +75,41 @@ def fixture_validate_options(request) -> None:
         msg = "When using a server CA, an external registry or rest URI must also be provided."
         raise ValueError(msg)
 
-    if has_external_registry_or_rest and not bootstrap_servers:
+    if has_external_registry_or_rest and not kafka_bootstrap_servers:
         msg = "When using an external registry or rest, the kafka bootstrap URIs must also be provided."
         raise ValueError(msg)
 
+    if not re.match(VERSION_REGEX, kafka_version):
+        msg = "Provided Kafka version has invalid format {kafka_version} should match {VERSION_REGEX}"
+        raise ValueError(msg)
+
+
+@pytest.fixture(scope="session", name="session_datadir")
+def fixture_session_datadir(tmp_path_factory) -> Path:
+    """Data files generated throught the tests should be stored here.
+
+    These files are NOT persisted.
+    """
+    return tmp_path_factory.mktemp("data")
+
+
+@pytest.fixture(scope="session", name="session_logdir")
+def fixture_session_logdir(request, tmp_path_factory) -> Path:
+    """All useful log data for debugging should be stored here.
 
-@pytest.fixture(scope="session", name="session_tmppath")
-def fixture_session_tmppath(tmp_path_factory) -> Path:
-    return tmp_path_factory.mktemp("karapace")
+    These files are persisted by the CI for debugging purposes.
+    """
+    log_dir = request.config.getoption("log_dir")
```

Comment:
locally we can just have the data and log files in the same directories, but for the CI the logs and Kafka's data must be in separate folders, otherwise the artifacts will be a couple of GB in size (Kafka pre allocate the files)